### ISSUE TITLE
[import/no-unused-modules] Doesn't detect unused exports through an export all

### ISSUE THREAD

**Issue by jquense** [import-js/eslint-plugin-import#2688]
*I'd be happy to contribute the fix if wanted*

I'd expect `n.js` to report it's export is unused, but it doesn't:

```
import { u1 } from './reexport'
```
```
//reexport.js`

export * from './n'
export * from './u'
```
```
// n.js
export const n1 = 'n
```
```
// u.js
export const u1 = 1
```

I've looked into trying to address this but it was more complicated than initially expected so i wanted to check in before diving down some road.  All the approaches i've initially tried end up being limited by how exportList tracks `ExportAllDeclarations`, in that it only marks them as used if it was directly used by an importing module, but not if there is a re-export in between. AFAICT there wasn't an obvious place to extend to add this, but i assume somewhere in the importList/exportList creation?

**Comment by lcwlucky** [comment]
I also met the same issue. Is there any solution？

**Comment by lluchkaa** [comment]
I also faced this issue. Will be glad to have a fix or a workaround here

**Comment by ljharb** [comment]
@jquense sorry this fell through the cracks; a PR would be most appreciated.

"export all" stuff is indeed tricky; it's handled inside ExportMap, I believe.
